homeassistant/components/zamg/config_flow.py

```diff
@@ -0,0 +1,87 @@
+"""Config Flow for zamg the Austrian "Zentralanstalt für Meteorologie und Geodynamik" integration."""
+from typing import Any
+
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.data_entry_flow import FlowResult
+
+from .const import CONF_STATION_ID, DOMAIN, LOGGER
+from .sensor import ZamgData, closest_station
+
+
+class ZamgConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Config flow for zamg integration."""
+
+    VERSION = 1
+
+    _client: ZamgData
+
+    def _show_setup_form(self, user_input=None, errors=None, station_id: str = ""):
+        """Show the setup form to the user."""
+        if user_input is None:
+            user_input = {}
+
+        return self.async_show_form(
+            step_id="user",
+            data_schema=vol.Schema(
+                {
+                    vol.Required(
+                        CONF_STATION_ID,
+                        default=user_input.get(CONF_STATION_ID, station_id),
+                    ): str
+                }
+            ),
+            errors=errors or {},
+        )
+
+    async def async_step_user(self, user_input=None):
+        """Handle a flow initiated by the user."""
+        errors = {}
+
+        station_id = await self.hass.async_add_executor_job(
+            closest_station,
+            self.hass.config.latitude,
+            self.hass.config.longitude,
+            self.hass.config.config_dir,
+        )
+
+        if user_input is None:
+            return self._show_setup_form(user_input, errors, station_id)
```

Comment:
`errors` is always an empty `dict`, right? Wouldn't it be nice to show an error to the user when the station id is invalid instead of aborting right away?

Reply:
Yes, that's right..how could i forgot..